### PR TITLE
KISS principle & fluent builder pattern

### PR DESCRIPTION
**Pull request by aminsharifi**
If it's good practice, may I do the same for the rest of the project's setters?

### CODE REVIEW

File: sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs [hunk 1]
```diff
@@ -5,15 +5,21 @@ namespace NimblePros.SampleToDo.Core.ContributorAggregate;
...
+    SetName(name);
   }
 
   public void UpdateName(string newName)
   {
-    Name = Guard.Against.NullOrEmpty(newName, nameof(newName));
+    SetName(newName);
+  }
+
+  public Contributor SetName(string newName)
```
Comment (ParsaMehdipour) [inline sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs:20]: When would we use a fluent builder for entities?

**Review by ParsaMehdipour — APPROVED** [review]

File: sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs (hunk 1 above)
Comment (aminsharifi) [reply sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs:20]: 1- For setting multiple properties at the same time.
2- Preventing multiple overloaded methods

File: sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
```
Comment (ardalis) [inline sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs:17]: Either UpdateName or SetName should be private/protected, and I would argue it's SetName. Is there a reason you left it public?

**Review by ardalis — CHANGES_REQUESTED** [review]

File: sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs (hunk 2 above)
Comment (aminsharifi) [reply sample/src/NimblePros.SampleToDo.Core/ContributorAggregate/Contributor.cs:17]: Above code is just for sample.
for replacement, I suppose this class:

﻿namespace NimblePros.SampleToDo.Core.ContributorAggregate;

public class Contributor : EntityBase, IAggregateRoot
{
  public string Name { get; private set; } = default!;

  public Contributor(string name)
  {
    UpdateName(name);
  }

  public Contributor UpdateName(string newName)
  {
    this.Name = Guard.Against.NullOrEmpty(newName, nameof(newName));
    return this;
  }
}

**Comment by KyleMcMaster** [comment]
I'm just curious as I try to understand the changes being proposed here. How does the KISS principle apply to having SetSomeProperty methods (potentially many methods if you're doing them individually)? Doesn't introducing setter methods make the Constructor, which itself is a setter, more complex?

**Review by KyleMcMaster — COMMENTED** [review]
I'm just curious as I try to understand the changes being proposed here. How does the KISS principle apply to having SetSomeProperty methods (potentially many methods if you're doing them individually)? Doesn't introducing setter methods make the Constructor, which itself is a setter, more complex?

**Comment by aminsharifi** [comment]
> I'm just curious as I try to understand the changes being proposed here. How does the KISS principle apply to having SetSomeProperty methods (potentially many methods if you're doing them individually)? Doesn't introducing setter methods make the Constructor, which itself is a setter, more complex?

Hello Kyle,
Thanks for your review
I wanted to bring to your attention that three commits have been made with the intention of following the KISS, DRY, and CQRS principles. Regarding the first commit you mentioned, I agree that it may not fully adhere to the KISS principle, but it does demonstrate the DRY principle. Specifically, the line `Guard.Against.NullOrEmpty(name, nameof(name))` appears to be duplicated in both the constructor and the UpdateName method.

**Review by ardalis — APPROVED** [review]

**Comment by ardalis** [comment]
I'm going to pull these in for now. My goal regarding checking Name in multiple places is to fix that using a Value Object rather than using a primitive, and then the check will live in the Value Object (only). But this will work for now and overall the other changes are helpful.

**Pull request merged**